moveit_core/robot_trajectory/include/moveit/robot_trajectory/robot_trajectory.h

```diff
@@ -204,7 +212,7 @@ class RobotTrajectory
 
   double getAverageSegmentDuration() const;
 
-  void getRobotTrajectoryMsg(moveit_msgs::msg::RobotTrajectory& trajectory) const;
+  void getRobotTrajectoryMsg(moveit_msgs::msg::RobotTrajectory& trajectory, rclcpp::Clock clock_ros_) const;
```

Comment:
if you are creating a attribute clock_ros_ you don't need to pass it as an argument

Reply:
I was having this error: 
```
error: passing ‘const rclcpp::Clock’ as ‘this’ argument discards qualifiers [-fpermissive]
   builtin_interfaces::msg::Time stamp = clock_ros_.now();
```